src/client/d3/d3.js

```diff
@@ -217,16 +217,30 @@ function render(updatedData) {
 
   $('#memory-input__submit').click((e) => {
     e.preventDefault();
-    $.ajax({
-      method: 'POST',
-      url: 'memory-input-text',
-      data: $('#add-text-form').serialize(),
-      success: () => {
-        setTimeout(() => {
-          update();
-        }, animationDuration);
-      },
-    });
+    if (navigator.onLine) {
+      $.ajax({
+        method: 'POST',
+        url: 'memory-input-text',
+        data: $('#add-text-form').serialize(),
+        success: () => {
+          setTimeout(() => {
+            update();
+          }, animationDuration);
+        },
+      });
+    } else {
+      const data = $('#add-text-form').serialize();
+      const splitData = data.split('&');
+      const heading = splitData[0].split('=')[1];
+      const text = splitData[1].split('=')[1];
+      const tag = splitData[2].split('=')[1];
+      const id = getRandomInt(10000, 999999);
```

Comment:
This is just temporary - a real id is created in the normal way once the user goes back online